server/live_query/redis_live_query.go

```diff
@@ -41,17 +59,111 @@ const (
 
 type redisLiveQuery struct {
 	// connection pool
-	pool *redisc.Cluster
+	pool fleet.RedisPool
 }
 
 // NewRedisQueryResults creats a new Redis implementation of the
 // QueryResultStore interface using the provided Redis connection pool.
-func NewRedisLiveQuery(pool *redisc.Cluster) *redisLiveQuery {
+func NewRedisLiveQuery(pool fleet.RedisPool) *redisLiveQuery {
 	return &redisLiveQuery{pool: pool}
 }
 
+// generate keys for the bitfield and sql of a query - those always go in pair
+// and should live on the same cluster node when Redis Cluster is used, so
+// the common part of the key (the 'name' parameter) is used as key tag.
 func generateKeys(name string) (targetsKey, sqlKey string) {
-	return queryKeyPrefix + name, sqlKeyPrefix + queryKeyPrefix + name
+	keyTag := "{" + name + "}"
```

Comment:
I'm assuming that we don't already have keys with "{}" in the name - which from what I've seen is a safe assumption, as RunQuery is only called from `server/service/service_campaigns.go` and is called with a (stringified) integer as name (`campaign.ID`).